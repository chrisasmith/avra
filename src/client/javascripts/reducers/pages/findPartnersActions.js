import { api } from '../../services'
import { createRequests, doApiCall, createRequestTypes } from '../actionHelper'

export const GET_FIND_PARTNERS_PAGE = createRequestTypes('GET_FIND_PARTNERS_PAGE')
export const getFindPartnersPage = createRequests(GET_FIND_PARTNERS_PAGE)

export const FIND_PARTNERS_PAGE_DATA_SUCCESS = 'FIND_PARTNERS_PAGE_DATA_SUCCESS'
export function findPartnersPageDataSuccess(data) {
  return { type: FIND_PARTNERS_PAGE_DATA_SUCCESS, data }
}

export function getFindPartnersData() {
  return doApiCall({
    api: api.getFindPartnersData,
    action: getFindPartnersPage,
  })
}
