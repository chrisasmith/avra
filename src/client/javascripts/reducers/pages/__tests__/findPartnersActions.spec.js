import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import fetchMock from 'fetch-mock'
import * as actions from '../findPartnersActions'

const mockStore = configureMockStore([thunk])

describe('FindPartners actions', () => {
  afterEach(() => {
    fetchMock.restore()
  })

  it('should call get find partners data', () => {
    fetchMock.mock('https://fake/findPartners.json',
      { body: { data: [] } }
    )

    const store = mockStore({})

    return store.dispatch(actions.getFindPartnersData())
      .then(() => {
        expect(store.getActions()).toMatchSnapshot()
      })
  })
})
