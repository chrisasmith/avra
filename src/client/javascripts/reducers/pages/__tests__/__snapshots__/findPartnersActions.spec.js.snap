// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindPartners actions should call get find partners data 1`] = `
Array [
  Object {
    "request": Object {},
    "type": "GET_FIND_PARTNERS_PAGE_REQUEST",
  },
  Object {
    "request": Object {},
    "response": Object {
      "data": Array [],
    },
    "type": "GET_FIND_PARTNERS_PAGE_SUCCESS",
  },
]
`;
